pennylane/templates/layers/basic_entangler.py

```diff
@@ -157,14 +122,91 @@ def circuit(weights):
         ``ValueError: Wrong number of parameters``.
     """
 
-    if rotation is None:
-        rotation = RX
+    num_params = 1
+    num_wires = AnyWires
+    par_domain = "A"
+
+    def __init__(self, weights, wires=None, rotation=None, do_queue=True):
```

Comment:
Another question, although maybe this was discussed in the ADR. Do we have a plan for supporting custom gradients?

Reply:
`decompose` is the old way to do it, and `expand` the way forward. In fact, in tape mode `expand` is called by PL, and expand then uses `decompose` where necessary to support the old version. 

And the important methods are `expand` but also `preprocess` where - depending on the template -  a lot of stuff can happen. The best example is `Motonnen` where there is tons of classical pre-processing of the features. 

But if a template does not need preprocessing, this method could be left out, it is not a required attribute of the class.

In contrast to before, the only difference really is the init method. If we rewrite `Operations` it may one day become even easier...